File: Data Structures & Algorithms/majority-element/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        let n = nums.length;
        for (let num of nums) {
            let count = nums.reduce((acc, val) => acc + (val === num ? 1 : 0), 0);
            if (count > Math.floor(n / 2)) {
                return num;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/majority-element-ii/submission-2.cs
public class Solution {
    public List<int> MajorityElement(int[] nums) {
        HashSet<int> res = new HashSet<int>();
        int n = nums.Length;

        foreach (int num in nums) {
            int count = nums.Count(x => x == num);
            if (count > n / 3) {
                res.Add(num);
            }
        }

        return res.ToList();
    }
}